third_party/blink/renderer/modules/encryptedmedia/media_keys.cc

```diff
@@ -443,12 +443,20 @@ ScriptPromise MediaKeys::getStatusForPolicy(
 
 #if BUILDFLAG(USE_STARBOARD_MEDIA)
 WebString MediaKeys::getMetrics(ExceptionState& exception_state) {
+  if (!cdm_) {
+    LOG(WARNING) << "Calling MediaKeys::getMetrics without a CDM";
+    exception_state.ThrowDOMException(
+        DOMExceptionCode::kInvalidStateError,
+        "Calling MediaKeys::getMetrics without a CDM");
+    return WebString();
+  }
   std::string metrics;
   if (cdm_->GetMetrics(metrics)) {
     return WebString::FromUTF8(metrics);
   }
   exception_state.ThrowDOMException(DOMExceptionCode::kInvalidStateError,
                                     "CDM returned empty GetMetrics()");
+  return WebString();
```

Comment:
Since in both cases the exception is the same, could we do something like
```
  std::string metrics;
  if (!cdm || !cdm->GetMetrics(metrics)) {
    exception_state.ThrowDOMException(
        DOMExceptionCode::kInvalidStateError,
        !!cdm ? "No active CDM" : "GetMetrics() failed");
    return WebString();
  }
  return metrics;
```

WDYT? Some people hate the ternary operator but I think in 
this case it would simplify the code here.